src/components/Select/index.js

```diff
@@ -1,36 +1,172 @@
-import React, { useState } from 'react';
+import React, { useContext, useEffect, useState } from 'react';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
-import { faCaretDown } from '@fortawesome/free-solid-svg-icons';
+import {
+  faCaretDown,
+  faEdit,
+  faPlus,
+  faTrashAlt
+} from '@fortawesome/free-solid-svg-icons';
+import swal from '@sweetalert/with-react';
 import { classes } from '../../utils';
 import { Button } from '..';
 import './stylesheet.scss';
+import { ThemeContext } from '../../contexts';
 
-export default function Select({ className, value, onChange, options }) {
+export default function Select({ className, value, options }) {
   const [opened, setOpened] = useState(false);
+  const [inputIndex, setInputIndex] = useState('');
+  const [inputting, setInputting] = useState(false);
+  const [versionName, setVersionName] = useState('');
```

Comment:
A lot of these options seem to specifically mention versionName (and versions) despite this being a general-purpose component. It would probably be better suited to be pulled out and become a specialized VersionSelect that includes these parts (in addition to the base Select code).

Reply:
This versionName variable is actually whatever the inputting value is. I just referenced it as versionName because I was working with verionName only. It is not tied with the actual versionName context and version cookie. My bad; I am renaming it to inputValue.